src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -1199,6 +1199,18 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		options.overflowWidgetsDomNode?.classList.add('hideSuggestTextIcons');
 		this._inputEditorElement.classList.add('hideSuggestTextIcons');
 
+		// Prevent Enter key from creating new lines when input is empty
+		this._register(this._inputEditor.onKeyDown((e) => {
+			if (e.keyCode === KeyCode.Enter && !e.altKey && !e.ctrlKey && !e.metaKey && !e.shiftKey) {
+				const model = this._inputEditor.getModel();
+				const inputHasText = !!model && model.getValue().trim().length > 0;
```

Comment:
Hm, just ocurred to me, maybe this `inputHasText` isn't needed because maybe the action will handle the enter key when active and this won't run anyway? I'm not sure

Reply:
But if not, we would also need to check the case when a request is currently in progress

Reply:
As I just validated, simply swallowing the 'enter' key breaks the normal scenario: user cannot hit enter even when the action is active.

It seems the chatInputPart handle the keyboard event earlier than the action..

Reply:
The keyboard event processing order is:
1. chatInputPart._inputEditor.onKeyDown
2. keybinding service (AbstractKeybindingService._dispatch)
3. monaco editor

so we cannot simply swallow the event at 1.

Reply:
Nevermind, I figured out a work around. Updated.